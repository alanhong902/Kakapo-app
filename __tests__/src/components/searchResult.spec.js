import React from 'react';
import { ImportSearchResult as SearchResult } from 'components/';
import { createComponentWithIntl } from '../helper';

const sound = {
  desc: 'Testing description',
  duration: '03:48',
  img: 'https://i.ytimg.com/vi/YTg7fpGLsKE/hqdefault.jpg',
  name: 'Testing name',
  tags: '',
  videoId: 'YTg7fpGLsKE',
  viewCount: 319609,
};

test('<SearchResult/> render youtube', () => {
  const props = {
    service: 'youtube',
    sound,
  };
  const tree = createComponentWithIntl(<SearchResult {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('<SearchResult/> render soundcloud', () => {
  const props = {
    service: 'soundcloud',
    sound,
  };
  const tree = createComponentWithIntl(<SearchResult {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
