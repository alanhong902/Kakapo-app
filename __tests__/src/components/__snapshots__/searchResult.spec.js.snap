// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResult/> render soundcloud 1`] = `
<div
  className="soundcloud-item"
  onClick={[Function]}
  role="button"
>
  <div
    className="thumbnail"
  >
    <img
      alt="Testing name"
      src="https://i.ytimg.com/vi/YTg7fpGLsKE/hqdefault.jpg"
    />
    <span
      className="duration"
    >
      03:48
    </span>
  </div>
  <span
    className="title"
  >
    Testing name
    <span
      className="view-count"
    >
      <span>
        319,609
      </span>
       
      <span>
        plays
      </span>
    </span>
  </span>
</div>
`;

exports[`<SearchResult/> render youtube 1`] = `
<div
  className="youtube-item"
  onClick={[Function]}
  role="button"
>
  <div
    className="thumbnail"
  >
    <img
      alt="Testing name"
      src="https://i.ytimg.com/vi/YTg7fpGLsKE/hqdefault.jpg"
    />
    <span
      className="duration"
    >
      03:48
    </span>
  </div>
  <span
    className="title"
  >
    Testing name
    <span
      className="view-count"
    >
      <span>
        319,609
      </span>
       
      <span>
        views
      </span>
    </span>
  </span>
</div>
`;
